py-polars/polars/io/database.py

```diff
@@ -184,4 +195,4 @@ def _open_adbc_connection(connection_uri: str) -> Any:
     if driver_name in ("sqlite", "snowflake"):
         connection_uri = re.sub(f"^{driver_name}:/{{,3}}", "", connection_uri)
 
-    return adbc_driver.connect(connection_uri)
+    return adbc_driver.connect(connection_uri, db_kwargs=db_kwargs)
```

Comment:
Passing `db_kwargs=db_kwargs` here isn't valid, as "db_kwargs" isn't a parameter on all of the `connect` methods (specifically, the SQLite driver does not have it).

As a result it gets passed down into a different `**kwargs` which finds itself here:
* [adbc_driver_sqlite/dbapi.py#L103](https://github.com/apache/arrow-adbc/blob/b97e22c4d6524b60bf261e1970155500645be510/python/adbc_driver_sqlite/adbc_driver_sqlite/dbapi.py#L103)

Now, the error is misleading because they have a bug in their exception handling :)

Rather than raising the _actual_ original error they call `.close()` in the except block, but because the `__init__` never completed the `__del__` checks for `_closed` - which the `__init__` never created - and raises a new error about that... :)

I'll probably submit a patch to fix their error handling here, but that looks like the root cause.

Reply:
TLDR: can't pass "db_kwargs" into sqlite adbc driver.

Reply:
I see what you mean.

I'm wondering if it might be best to close this PR, and instead handle all this by passing a `Connection` object as suggested in #10121. Then we don't have to worry about any of these implementation details.

Reply:
Very good catch, btw. Had me scratching my head!

Reply:
> I'm wondering if it might be best to close this PR, and instead handle all this by passing a `Connection` object as suggested in #10121. Then we don't have to worry about any of these implementation details.

I was thinking the same :) Allowing connections to be passed "as-is" will allow you to establish the connection you need (and potentially reuse it) without having to pass around a pile of kwargs to polars. 

Once I've stopped disassembling lambdas (https://github.com/pola-rs/polars/pull/10085) and staring at the resulting bytecode I think this will be the next priority, so you shouldn't have to wait long. Feels like a good opportunity to further expand polars' connectivity options...